### PR TITLE
[Scala] Extract concepts of rna_transcription exercise 

### PR DESCRIPTION
**Pull request by outkaj**

### CODE REVIEW

**Review by ErikSchierboom — COMMENTED** [review]
This is looking great! Some Concepts that might be added too:

- generics (for option definition)
- strings: input parameter
- immutability: use `val` to define mapping

File: languages/scala/reference/exercise-concepts/rna-transcription.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+## General
+
+- object or case class: used as class wrapper for exercises
+- functions: used as main entry point for the exercise
+- function arguments: input strands passed as arguments
+- return values: return a value from a function
+- Char: single character, manipulate text
+- parameter scope: use of `private`
+- foldRight: collapse a sequence from inside to outside
```
Comment (leobenkel) [inline languages/scala/reference/exercise-concepts/rna-transcription.md]: This is under the `collection - reduce` main concept i think.

File: languages/scala/reference/exercise-concepts/rna-transcription.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+- pattern matching: handle a value differently if it fits a specific pattern
+- for comprehension: yield a sequence matching certain conditions
+- Map: to store key/value pairs
+- Option: `Option` is used as return result
```
Comment (leobenkel) [inline languages/scala/reference/exercise-concepts/rna-transcription.md]: Maybe something about `null handling` ?

**Review by leobenkel — CHANGES_REQUESTED** [review]

**Comment by ErikSchierboom** [comment]
@outkaj Is there anything I can do to help with this PR?

**Comment by ErikSchierboom** [comment]
This Friday (Jan 29th 2021), we're going to explode this repo back into the individual track repositories and archive this repo.

As such, this PR needs to be merged by Friday 9am UTC, or it will need to be manually reopened against the track repo.

**Pull request closed without merging**